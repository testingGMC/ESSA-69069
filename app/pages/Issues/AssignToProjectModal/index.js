/**
 *
 * AssignToProjectModal
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { DAEMON } from 'utils/constants'
import { useInjectReducer } from 'utils/injectReducer'

import makeSelectSprints from 'shared/redux/sprints/selectors'
import reducer from 'shared/redux/sprints/reducer'
import * as actions from 'shared/redux/sprints/actions'
import saga from 'shared/redux/sprints/saga'

import AssignToProjectModal from './AssignToProjectModal'

const AssignToProjectModalIndex = props => {
  useInjectReducer({ key: 'sprints', reducer })

  return <AssignToProjectModal {...props} />
}

const mapStateToProps = createStructuredSelector({
  sprints: makeSelectSprints(),
})

const mapDispatchToProps = {
  ...actions,
}

const withSaga = injectSaga({ key: 'sprints', saga, mode: DAEMON })

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  memo,
  withSaga,
)(AssignToProjectModalIndex)
