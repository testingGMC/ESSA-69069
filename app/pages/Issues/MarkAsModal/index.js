import { Form } from 'antd'
import MarkAsModal from './MarkAsModal'
export default Form.create({ name: 'mark_as' })(MarkAsModal)
