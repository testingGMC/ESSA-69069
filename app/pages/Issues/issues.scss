.issues {
    padding: 27px 23px 47px 35px;

    .row-title {
        align-items: center;
        margin-bottom: 27px;

        .ant-page-header {
            display: inline-flex;
            align-items: center;
            padding: 16px 4px;

            .ant-page-header-title-view-title {
                font-size: 28px;
            }
        }

        .issues-filter {
            text-align: right;

            .filter-by {
                color: #aaaeb3;
                margin-right: 10px;
            }



            .ant-select-selection {
                background-color: transparent !important;
                border: none !important;

                .ant-select-selection__placeholder {
                    top: 61% !important;
                    right: 4px !important;
                    left: -18px !important;
                    text-align: right !important;
                    font-size: 13px;
                    font-weight: 600;
                }

                .ant-select-selection-selected-value {
                    color: #bfbfbf;
                    top: 11% !important;
                    right: 4px !important;
                    left: -18px !important;
                    text-align: right !important;
                    position: absolute;
                    padding-right: 0px;
                }
            }

        }
    }

    .ant-table-thead>tr,
    .ant-table-tbody>tr {
        background: white;
    }

    .ant-table-column-title {
        color: #aaaeb3;
        font-size: 14px;
    }



}