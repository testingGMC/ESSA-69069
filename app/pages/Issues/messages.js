/*
 * Issues Messages
 *
 * This contains all the text for the Issues container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.Issues'

export default defineMessages({
  issues: {
    id: `${scope}.issues`,
    defaultMessage: 'Issues',
  },
  filterBy: {
    id: `${scope}.filterBy`,
    defaultMessage: 'Filter by',
  },

  acknowledged: {
    id: `${scope}.issues`,
    defaultMessage: 'Acknowledged',
  },
  rejected: {
    id: `${scope}.rejected`,
    defaultMessage: 'Rejected',
  },
  duplicated: {
    id: `${scope}.duplicated`,
    defaultMessage: 'Duplicated',
  },

  resolved: {
    id: `${scope}.resolved`,
    defaultMessage: 'Resolved',
  },
  pending: {
    id: `${scope}.pending`,
    defaultMessage: 'Pending',
  },
  allSkills: {
    id: `${scope}.allSkills`,
    defaultMessage: 'All Skills',
  },
  markAs: {
    id: `${scope}.markAs`,
    defaultMessage: 'Mark As',
  },
  assignTo: {
    id: `${scope}.assignTo`,
    defaultMessage: 'Assign to',
  },
  status: {
    id: `${scope}.status`,
    defaultMessage: 'Status',
  },
  name: {
    id: `${scope}.name`,
    defaultMessage: 'Name',
  },
  description: {
    id: `${scope}.description`,
    defaultMessage: 'Description',
  },
  date: {
    id: `${scope}.date`,
    defaultMessage: 'Date',
  },
  reporter: {
    id: `${scope}.reporter`,
    defaultMessage: 'Reporter',
  },
  actions: {
    id: `${scope}.actions`,
    defaultMessage: 'Actions',
  },
  cancel: {
    id: `${scope}.cancel`,
    defaultMessage: 'Cancel',
  },
  validate: {
    id: `${scope}.validate`,
    defaultMessage: 'Validate',
  },
  workItemName: {
    id: `${scope}.workItemName`,
    defaultMessage: 'Work item name',
  },
  workItemDescription: {
    id: `${scope}.workItemDescription`,
    defaultMessage: 'Work item description',
  },
  project: {
    id: `${scope}.project`,
    defaultMessage: 'Project name',
  },
  sprint: {
    id: `${scope}.sprint`,
    defaultMessage: 'Sprint name',
  },
  statusError: {
    id: `${scope}.statusError`,
    defaultMessage: 'Please input the status!',
  },
  projectError: {
    id: `${scope}.projectError`,
    defaultMessage: 'Please input the project!',
  },
  sprintError: {
    id: `${scope}.sprintError`,
    defaultMessage: 'Please input the sprint!',
  },
  workItemNameError: {
    id: `${scope}.workItemNameError`,
    defaultMessage: 'Please input the work item name!',
  },
  workItemDescriptionError: {
    id: `${scope}.workItemDescriptionError`,
    defaultMessage: 'Please input the work item description',
  },
  helmetDescription: {
    id: `${scope}.helmetDescription`,
    defaultMessage: 'Description of Issues',
  },
  helmetTitle: {
    id: `${scope}.helmetTitle`,
    defaultMessage: 'Issues',
  },
})
