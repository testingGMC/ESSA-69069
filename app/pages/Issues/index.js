/**
 *
 * Issues
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'

import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { DAEMON } from 'utils/constants'
import { useInjectReducer } from 'utils/injectReducer'

import makeSelectIssues from 'shared/redux/issues/selectors'
import reducer from 'shared/redux/issues/reducer'
import * as actions from 'shared/redux/issues/actions'
import saga from 'shared/redux/issues/saga'

import makeSelectSkills from 'shared/redux/skills/selectors'
import skillsReducer from 'shared/redux/skills/reducer'
import * as skillsActions from 'shared/redux/skills/actions'
import skillsSaga from 'shared/redux/skills/saga'

import makeSelectSprints from 'shared/redux/sprints/selectors'
import sprintsReducer from 'shared/redux/sprints/reducer'
import { addWorkItem } from 'shared/redux/sprints/actions'
import sprintsSaga from 'shared/redux/sprints/saga'

import makeSelectProjects from 'shared/redux/projects/selectors'
import projectsReducer from 'shared/redux/projects/reducer'
import { fetchProjects } from 'shared/redux/projects/actions'
import projectsSaga from 'shared/redux/projects/saga'

import Issues from './Issues'

const IssuesIndex = props => {
  useInjectReducer({ key: 'issues', reducer })
  useInjectReducer({ key: 'skills', reducer: skillsReducer })
  useInjectReducer({ key: 'sprints', reducer: sprintsReducer })
  useInjectReducer({ key: 'projects', reducer: projectsReducer })

  return <Issues {...props} />
}

const mapStateToProps = createStructuredSelector({
  issues: makeSelectIssues(),
  skills: makeSelectSkills(),
  sprints: makeSelectSprints(),
  projects: makeSelectProjects(),
})

const mapDispatchToProps = {
  ...actions,
  ...skillsActions,
  addWorkItem,
  fetchProjects,
}

const withSaga = injectSaga({ key: 'issues', saga, mode: DAEMON })
const withSkillsSaga = injectSaga({
  key: 'skills',
  saga: skillsSaga,
  mode: DAEMON,
})
const withSprintsSaga = injectSaga({
  key: 'sprints',
  saga: sprintsSaga,
  mode: DAEMON,
})

const withProjectsSaga = injectSaga({
  key: 'projects',
  saga: projectsSaga,
  mode: DAEMON,
})
const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  memo,
  withSaga,
  withSkillsSaga,
  withSprintsSaga,
  withProjectsSaga,
)(IssuesIndex)
