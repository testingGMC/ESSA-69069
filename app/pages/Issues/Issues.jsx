/**
 *
 * Issues
 *
 */

import React, { useEffect, useState } from 'react'
import { Table, Row, Col, PageHeader } from 'antd'
import PropTypes from 'prop-types'
import { injectIntl, FormattedMessage, intlShape } from 'react-intl'
import { Helmet } from 'react-helmet'
import messages from './messages'
import TagFilter from './TagFilter'
import MarkAsModal from './MarkAsModal'
import useActions from './hooks/useActions'

import './issues.scss'
import AssignToProjectModal from './AssignToProjectModal'

const Issues = ({
  issues,
  fetchIssues,
  fetchProjects,
  projects,
  updateStatus,
  addWorkItem,
  intl,
}) => {
  const [statusValue, setStatusValue] = useState(-1)
  const [issuesData, setIssuesData] = useState(issues.data ? issues.data : [])
  const record = useActions()
  const [formRef, setFormRef] = useState(null)

  useEffect(() => {
    fetchIssues()
    fetchProjects()
  }, [])

  useEffect(() => {
    let issuesDataVar = []
    if (statusValue !== -1) {
      issuesDataVar = issues.data
        ? issues.data.filter(i => i.status === statusValue)
        : []
    } else {
      issuesDataVar = issues.data
    }

    setIssuesData(issuesDataVar)
  }, [issues, statusValue])
  const handleStatusValue = title => {
    if (title === statusValue) {
      setStatusValue(-1)
    } else {
      setStatusValue(title)
    }
  }
  const handleMarkAs = () => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      form.resetFields()

      const issue = {
        status: values.status,
      }
      updateStatus(record.issueId, issue)
      handleMarkAsModalClose()
    })
  }
  const handleAssignToProject = () => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      form.resetFields()
      const workItem = {
        ...values,
        issueId: record.issueId,
        type: 1,
        state: 0,
      }
      addWorkItem(workItem)
      handleAssignToProjectModalClose()
    })
  }
  const handleMarkAsModalClose = () => {
    record.setIsMarkAsModalVisible(false)
  }
  const handleAssignToProjectModalClose = () => {
    record.setIsAssignToProjectModalVisible(false)
  }
  const saveFormRef = formRefParameter => {
    setFormRef(formRefParameter)
  }
  const helmetMessages = {
    helmetTitle: intl.formatMessage({
      ...messages.helmetTitle,
    }),
    helmetDescription: intl.formatMessage({
      ...messages.helmetDescription,
    }),
  }
  return (
    <div className="issues">
      <Helmet>
        <title>{helmetMessages.helmetTitle}</title>
        <meta name="description" content={helmetMessages.helmetDescription} />
      </Helmet>
      <Row gutter={24} type="flex" className="row-title">
        <Col xs={24} sm={24} md={8} lg={8} xl={8}>
          <PageHeader
            onBack={() => {
              window.history.back()
            }}
            title={<FormattedMessage {...messages.issues} />}
          />
        </Col>
        <Col xs={24} sm={24} md={16} lg={16} xl={16}>
          <div className="issues-filter">
            <span className="filter-by">
              <FormattedMessage {...messages.filterBy} />:
            </span>
            <TagFilter
              handleStatusValue={handleStatusValue}
              statusValue={statusValue}
            />
          </div>
        </Col>
      </Row>
      <Row gutter={24}>
        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
          <Table
            loading={issues.loading}
            columns={record.columns}
            dataSource={issuesData}
            scroll={{ x: 1000 }}
          />
        </Col>
      </Row>
      {record.isMarkAsModalVisible && (
        <MarkAsModal
          wrappedComponentRef={saveFormRef}
          isVisible={record.isMarkAsModalVisible}
          handleCancel={handleMarkAsModalClose}
          status={record.status}
          handleOk={handleMarkAs}
        />
      )}
      {record.isAssignToProjectModalVisible && (
        <AssignToProjectModal
          wrappedComponentRef={saveFormRef}
          isVisible={record.isAssignToProjectModalVisible}
          handleCancel={handleAssignToProjectModalClose}
          handleOk={handleAssignToProject}
          issueName={record.name}
          issueDescription={record.description}
          projects={projects.data ? projects.data : []}
        />
      )}
    </div>
  )
}

Issues.propTypes = {
  issues: PropTypes.object.isRequired,
  fetchIssues: PropTypes.func.isRequired,
  projects: PropTypes.object.isRequired,
  updateStatus: PropTypes.func.isRequired,
  addWorkItem: PropTypes.func.isRequired,
  fetchProjects: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
}

export default injectIntl(Issues)
