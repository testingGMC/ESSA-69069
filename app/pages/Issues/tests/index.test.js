import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory } from 'react-router-dom'

import IssuesIndex from '../index'
import Issues from '../Issues'
import { DEFAULT_LOCALE } from '../../../i18n'
import configureStore from '../../../configureStore'

const initialProps = {
  fetchIssues: jest.fn(),
  updateStatus: jest.fn(),
  issues: {},
  skills: {},
  projects: [],
  fetchSkills: jest.fn(),
  updateStatus: jest.fn(),
  addWorkItem: jest.fn(),
  fetchProjects: jest.fn(),
}

describe('<IssuesIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <IssuesIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <IssuesIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

})

describe('<Issues />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Issues {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()
  })

  it('should fetch the issues on mount if the issueId exists', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Issues {...initialProps} issues skills projects />
        </IntlProvider>
      </Provider>,
    )
    expect(initialProps.fetchIssues).toHaveBeenCalled()
  })
})