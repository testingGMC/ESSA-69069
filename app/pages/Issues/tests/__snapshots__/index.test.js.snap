// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Issues /> should render and match the snapshot 1`] = `
<div
  class="issues"
>
  <div
    class="ant-row-flex row-title"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-8 ant-col-lg-8 ant-col-xl-8"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="ant-page-header"
      >
        <div
          class="ant-page-header-back"
        >
          <div
            aria-label="back"
            class="ant-page-header-back-button"
            role="button"
            style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
            tabindex="0"
          >
            <i
              aria-label="icon: arrow-left"
              class="anticon anticon-arrow-left"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="arrow-left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </i>
          </div>
          <div
            class="ant-divider ant-divider-vertical"
          />
        </div>
        <div
          class="ant-page-header-title-view"
        >
          <span
            class="ant-page-header-title-view-title"
          >
            <span>
              Issues
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-16 ant-col-lg-16 ant-col-xl-16"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="issues-filter"
      >
        <span
          class="filter-by"
        >
          <span>
            Filter by
          </span>
          :
        </span>
        <span
          class="tag-filter"
        >
          <div
            class="ant-tag ant-tag-checkable tag-1 false"
            color="orange"
          >
            <span>
              Acknowledged
            </span>
          </div>
          <div
            class="ant-tag ant-tag-checkable tag-2 false"
            color="red"
          >
            <span>
              Rejected
            </span>
          </div>
          <div
            class="ant-tag ant-tag-checkable tag-3 false"
            color="cyan"
          >
            <span>
              Duplicated
            </span>
          </div>
          <div
            class="ant-tag ant-tag-checkable tag-4 false"
            color="green"
          >
            <span>
              Resolved
            </span>
          </div>
          <div
            class="ant-tag ant-tag-checkable tag-5 false"
            color="geekblue"
          >
            <span>
              Pending
            </span>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left ant-table-scroll-position-right"
            >
              <div
                class="ant-table-content"
              >
                <div
                  class="ant-table-scroll"
                >
                  <div
                    class="ant-table-body"
                    style="overflow-x: scroll;"
                  >
                    <table
                      class="ant-table-fixed"
                      style="width: 1000px;"
                    >
                      <colgroup>
                        <col
                          style="width: 100px; min-width: 100px;"
                        />
                        <col
                          style="width: 150px; min-width: 150px;"
                        />
                        <col
                          style="width: 40%; min-width: 40%;"
                        />
                        <col />
                        <col />
                        <col />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-fixed-columns-in-body ant-table-align-center"
                            style="text-align: center;"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  <span>
                                    Status
                                  </span>
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-fixed-columns-in-body ant-table-column-has-actions ant-table-column-has-sorters"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  <span>
                                    Name
                                  </span>
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                >
                                  <div
                                    class="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                    title="Sort"
                                  >
                                    <i
                                      aria-label="icon: caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                    <i
                                      aria-label="icon: caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                  </div>
                                </span>
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-sorters"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  <span>
                                    Description
                                  </span>
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                >
                                  <div
                                    class="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                    title="Sort"
                                  >
                                    <i
                                      aria-label="icon: caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                    <i
                                      aria-label="icon: caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                  </div>
                                </span>
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-sorters"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  <span>
                                    Date
                                  </span>
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                >
                                  <div
                                    class="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                    title="Sort"
                                  >
                                    <i
                                      aria-label="icon: caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                    <i
                                      aria-label="icon: caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                  </div>
                                </span>
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-sorters"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  <span>
                                    Reporter
                                  </span>
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                >
                                  <div
                                    class="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                    title="Sort"
                                  >
                                    <i
                                      aria-label="icon: caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                    <i
                                      aria-label="icon: caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                  </div>
                                </span>
                              </div>
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  <span>
                                    Actions
                                  </span>
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      />
                    </table>
                  </div>
                  <div
                    class="ant-table-placeholder"
                  >
                    <div
                      class="ant-empty ant-empty-normal"
                    >
                      <div
                        class="ant-empty-image"
                      >
                        <img
                          alt="No Data"
                          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                        />
                      </div>
                      <p
                        class="ant-empty-description"
                      >
                        No Data
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-table-fixed-left"
                >
                  <div
                    class="ant-table-body-outer"
                  >
                    <div
                      class="ant-table-body-inner"
                    >
                      <table
                        class="ant-table-fixed"
                      >
                        <colgroup>
                          <col
                            style="width: 100px; min-width: 100px;"
                          />
                          <col
                            style="width: 150px; min-width: 150px;"
                          />
                        </colgroup>
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-align-center"
                              style="text-align: center;"
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div>
                                  <span
                                    class="ant-table-column-title"
                                  >
                                    <span>
                                      Status
                                    </span>
                                  </span>
                                  <span
                                    class="ant-table-column-sorter"
                                  />
                                </div>
                              </span>
                            </th>
                            <th
                              class="ant-table-column-has-actions ant-table-column-has-sorters"
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div
                                  class="ant-table-column-sorters"
                                >
                                  <span
                                    class="ant-table-column-title"
                                  >
                                    <span>
                                      Name
                                    </span>
                                  </span>
                                  <span
                                    class="ant-table-column-sorter"
                                  >
                                    <div
                                      class="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                      title="Sort"
                                    >
                                      <i
                                        aria-label="icon: caret-up"
                                        class="anticon anticon-caret-up ant-table-column-sorter-up off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          class=""
                                          data-icon="caret-up"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                      <i
                                        aria-label="icon: caret-down"
                                        class="anticon anticon-caret-down ant-table-column-sorter-down off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          class=""
                                          data-icon="caret-down"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                    </div>
                                  </span>
                                </div>
                              </span>
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        />
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
